scripts/xmpro-website-scripts/scrape-xmpro-website-latestnews.py

```diff
@@ -0,0 +1,165 @@
+        try:
+            page = int(page_number.text)
+            page_url = urljoin(base_url, page_number.get('href'))
+            if page > max_page:
+                max_page = page
+        except ValueError:
+            pass  # Ignore non-integer page numbers
+    return max_page
+
+def generate_readme(files, folder_path):
+    try:
+        readme_content = []
+        for file_info in files:
+            # Remove "docs/" from file path and replace "\" with "/"
+            file_path = file_info['path'].replace("docs/", "").replace("\\", "/")
+            readme_content.append(f"* [{file_info['title']}]({file_path})\n")
+        
+        # Create the README.md file in the same folder as the exported files
+        readme_file_path = os.path.join(folder_path, 'copy-me-latest-news.md')
+        with open(readme_file_path, 'w', encoding='utf-8') as readme_file:
+            readme_file.write("".join(readme_content))
+        print(f"README.md file created successfully at: {readme_file_path}")
+    except Exception as e:
+        print(f"Error occurred while generating README.md: {e}")
+
+def main():
+    # Load configuration from JSON file
+    with open('scripts\XMPRO Website Scrape Scripts\scrape-xmpro-website-latestnews-config.json') as json_file:
+        config = json.load(json_file)
+
+    # Extract folder path from config
+    folder_path = config.get('folderPath')
+
+    if folder_path:
+        # Ensure the folder path exists, create if it doesn't
+        os.makedirs(folder_path, exist_ok=True)
+
+        # Example HTML snippet
+        html_snippet = '''
+        <ul class="page-numbers nav-pagination links text-center">
+            <li><span aria-current="page" class="page-number current">1</span></li>
+            <li><a class="page-number" href="/category/news/page/2/">2</a></li>
+            <li><a class="page-number" href="/category/news/page/3/">3</a></li>
+            <li><a class="page-number" href="/category/news/page/4/">4</a></li>
+            <li><span class="page-number dots">…</span></li>
+            <li><a class="page-number" href="/category/news/page/14/">14</a></li>
+            <li><a class="next page-number" href="/category/news/page/2/"><i class="icon-angle-right"></i></a></li>
+        </ul>
+        '''
+
+        base_url = "https://xmpro.com/category/news/"
+        num_pages = get_max_page_numbers(html_snippet, base_url)
+
+        all_blog_urls = get_all_blog_urls(base_url, num_pages)
+        exported_files = []
+
+        for url in all_blog_urls:
+            # Introduce a delay of 1 second before scraping each page
+            sleep(1)
+            title, content = scrape_page(url)
+            if title and content:
+                filename = save_to_md(title, content, url, folder_path)
+                if filename:
+                    exported_files.append({'title': title, 'path': filename})
+            else:
+                print(f"Failed to scrape the page: {url}")
```

Comment:
@jacksonmcguirkxmpro 

Lets consider what happens if `scrape_page(url) fails` we might want to either allow the `scape_page()` function to gracefully fail (not suddenly throw error, but instead return `False` or something) or wrap it up in a `try-except` statement to catch errors and log - then continue